include/slam_toolbox/loop_closure_assistant.hpp

```diff
@@ -45,24 +45,51 @@ class LoopClosureAssistant
 {
 public:
   LoopClosureAssistant(
-    rclcpp::Node::SharedPtr node, karto::Mapper * mapper,
-    laser_utils::ScanHolder * scan_holder, PausedState & state,
-    ProcessType & processor_type);
+    rclcpp::node_interfaces::NodeBaseInterface::SharedPtr base_interface,
```

Comment:
I fail to see why the second contructor is required or that we pass all the interfaces to it manually. If this is templated, use the templated node type